src/Plugin/SolrConnector/PantheonSolrConnector.php

```diff
@@ -399,18 +407,125 @@ public function getEndpoint($key = 'search_api_solr') {
   }
 
   /**
-   * Reaload the Solr Core.
-   *
-   * @return bool
-   *   Success or Failure.
-   */
-  public function reloadCore() {
-    $this->logger->notice(
-          $this->t('Reload Core action for Pantheon Solr is automatic when Schema is updated.')
-      );
-    return TRUE;
+ * Reloads the Solr Core with updated configuration.
+ *
+ * @return bool
+ *   TRUE if reload succeeded, FALSE otherwise.
+ */
+public function reloadCore() {
+  $maxRetries = 3;
+  $retryDelay = 2;
+  $attempt = 0;
+
+  do {
+    try {
+      $attempt++;
+      
+      // Log reload attempt
+      $this->logger->info('Attempting Solr core reload (attempt @attempt of @max) for core: @core', [
+        '@attempt' => $attempt,
+        '@max' => $maxRetries,
+        '@core' => $this->configuration['core']
+      ]);
+
+      // Make reload request
+      $response = $this->pantheonGuzzle->get('admin/cores', [
+        'query' => [
+          'action' => 'RELOAD',
+          'core' => $this->configuration['core'],
+          'wt' => 'json'
+        ],
+        'headers' => [
+          'Accept' => 'application/json'
+        ]
+      ]);
+
+      $result = json_decode($response->getBody(), TRUE);
+
+      if ($response->getStatusCode() !== 200) {
+        throw new SearchApiSolrException(sprintf(
+          'Solr core reload failed with status code %d: %s',
+          $response->getStatusCode(),
+          $result['error']['msg'] ?? 'Unknown error'
+        ));
+      }
+
+      // Verify core status after reload
+      $coreStatus = $this->verifyCoreStatus();
+      
+      if ($coreStatus) {
+        $this->logger->info('Successfully reloaded Solr core @core', [
+          '@core' => $this->configuration['core']
+        ]);
+        
+        $this->messenger->addMessage($this->t('Solr core configuration has been reloaded successfully.'));
+        return TRUE;
+      }
+
+      throw new SearchApiSolrException('Core reload completed but core status verification failed');
+
+    } catch (\Exception $e) {
+      $this->logger->error('Core reload attempt @attempt failed: @message', [
+        '@attempt' => $attempt,
+        '@message' => $e->getMessage()
+      ]);
+
+      if ($attempt < $maxRetries) {
+        sleep($retryDelay * $attempt); // Exponential backoff
+        continue;
+      }
+
+      $this->messenger->addError($this->t('Failed to reload Solr core after @max attempts. Please check logs for details.', [
+        '@max' => $maxRetries
+      ]));
+      return FALSE;
+    }
+  } while ($attempt < $maxRetries);
+
+  return FALSE;
+}
+
+/**
+ * Verifies core status after reload.
+ *
+ * @return bool
+ *   TRUE if core is active, FALSE otherwise.
+ */
+protected function verifyCoreStatus() {
+  $maxAttempts = 5;
+  $attempt = 0;
+  
+  while ($attempt < $maxAttempts) {
+    try {
+      $response = $this->pantheonGuzzle->get('admin/cores', [
```

Comment:
/admin/cores is inaccessible from the standard pantheon environment. We have a process for reloading the core.